websites/P/Paramount+/presence.ts

```diff
@@ -9,145 +9,179 @@ const presence = new Presence({
-			else if (path.includes("/video")) {
-				title = jsonData.partOfSeries.name;
-
-				seasonEpi =
-					`S${jsonData.partOfSeason.seasonNumber}:` +
-					`E${jsonData.episodeNumber} ${jsonData.name}`;
+		{ pathname: path } = document.location;
+
+	switch (true) {
+		case pathIncludes(path, "/home"):
+			presenceData.details = "Browsing";
+			presenceData.state = "Viewing home page";
+			break;
+
+		case pathIncludes(path, "/search"):
+			presenceData.details = "Searching";
+			presenceData.smallImageKey = Assets.Search;
+			presenceData.smallImageText = "Looking for something good";
+			break;
+
+		case pathIncludes(path, "/shows"): {
+			if (vidArea && pathIncludes(path, "/video")) {
+				const data = JSON.parse(
+					document.querySelector('[type="application/ld+json"]').textContent
+				);
+
+				video = document.querySelector("video");
+
+				title = data.partOfSeries.name;
+
+				seasonEpisode = `S${data.partOfSeason.seasonNumber}:E${data.episodeNumber} - ${data.name}`;
+
+				if (title) presenceData.details = title;
+
+				if (seasonEpisode) presenceData.state = seasonEpisode;
+
+				presenceData.smallImageKey = video.paused ? Assets.Pause : Assets.Play;
+				presenceData.smallImageText = video.paused
+					? (await strings).pause
+					: (await strings).play;
+				presenceData.largeImageKey = data.image || Logos.Paramount;
+
+				[presenceData.startTimestamp, presenceData.endTimestamp] =
+					presence.getTimestampsfromMedia(video);
+
+				if (video.paused) {
+					delete presenceData.startTimestamp;
+					delete presenceData.endTimestamp;
+				}
+			} else if (
+				!vidArea &&
+				document.querySelector('[type="application/ld+json"]') !== null
+			) {
+				presenceData.details = "Viewing series";
+				presenceData.state = JSON.parse(
+					document.querySelector('[type="application/ld+json"]').textContent
+				).name;
+			} else {
+				presenceData.details = "Browsing";
+				presenceData.state = "Viewing Shows";
 			}
+			break;
+		}
 
-			const content = seasonEpi;
-
-			if (movTitle) {
-				presenceData.state = movTitle;
-				presenceData.details = "Watching Movie:";
+		case pathIncludes(path, "/movies"):
+			if (
+				vidArea &&
+				document.querySelector("video.marqueeVideo") &&
+				!vidArea.querySelector("video")
+			) {
+				presenceData.details = "Previewing a movie";
+				presenceData.state = JSON.parse(
+					document.querySelector('[type="application/ld+json"]').textContent
+				).name;
+			} else if (vidArea && vidArea.querySelector("video")) {
+				const movData = JSON.parse(
+					document.querySelector('[type="application/ld+json"]').textContent
+				);
+
+				video = vidArea.querySelector("video");
+
+				if (movData.name) presenceData.state = movData.name;
+				presenceData.details = "Watching a movie";
+
+				presenceData.smallImageKey = video.paused ? Assets.Pause : Assets.Play;
+				presenceData.smallImageText = video.paused
+					? (await strings).pause
+					: (await strings).play;
```

Comment:
Await the strings at the top of the file, look at other presences like Youtube to see how this is done.